### PR TITLE
[clang] Implement CWG1719 "Layout compatibility and cv-qualification revisited"

### PR DESCRIPTION
**Pull request by Endilll**
This patch updates our internal notion of `layout-compatible`, which in turn fixes `__is_layout_compatible` intrinsic.

### CODE REVIEW

File: clang/test/CXX/drs/dr17xx.cpp [hunk 1]
```diff
@@ -66,11 +66,10 @@ struct CStructWithQualifiers {
...
 static_assert(__is_layout_compatible(CStruct, volatile CStruct2), "");
 static_assert(__is_layout_compatible(const CStruct, volatile CStruct2), "");
-// FIXME: all of the following pairs of types are layout-compatible
-static_assert(!__is_layout_compatible(int, const int), "");
-static_assert(!__is_layout_compatible(int, volatile int), "");
-static_assert(!__is_layout_compatible(const int, volatile int), "");
-static_assert(!__is_layout_compatible(CStruct, CStructWithQualifiers), "");
+static_assert(__is_layout_compatible(int, const int), "");
+static_assert(__is_layout_compatible(int, volatile int), "");
+static_assert(__is_layout_compatible(const int, volatile int), "");
```
[the remark this answers is not in the document]
Comment (Endilll) [reply clang/test/CXX/drs/dr17xx.cpp:71]: Added, thank you!
Save for the fact that I don't think `__restrict` and `_Atomic` belong in a C++ DR tests, so I put them in SemaCXX tests.

File: clang/docs/ReleaseNotes.rst [hunk 2]
```diff
@@ -120,6 +119,10 @@ Resolutions to C++ Defect Reports
   in the template parameters, but is deduced from a previous argument.
   (`#78449: <https://github.com/llvm/llvm-project/issues/78449>`_).
 
+- ``const`` and ``volatile`` qualifiers are now ignored when evaluating
```
[the remark this answers is not in the document]
Comment (Endilll) [reply clang/docs/ReleaseNotes.rst]: Done

[the remark this answers is not in the document]
Comment (Endilll) [reply clang/docs/ReleaseNotes.rst]: Done

**Review by cor3ntin — APPROVED** [review]

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Pull request merged**